src/utils/config_utils/src/config_utils.c

```diff
@@ -330,9 +389,126 @@ bool ADUC_ConfigInfo_Init(ADUC_ConfigInfo* config, const char* configFilePath)
         goto done;
     }
 
+    const char* aduShellFolder = ADUC_JSON_GetStringFieldPtr(root_value, CONFIG_ADU_SHELL_FOLDER);
+
+    if (aduShellFolder != NULL)
+    {
+        if (mallocAndStrcpy_s(&(config->aduShellFolder), aduShellFolder) != 0)
```

Comment:
why doing all this malloc and copy? Callers do not need clone of config file but can just share global obj. At very least, make clone_config() function.

Reply:
The new code follows the existing code pattern, where we copy strings from a JSON object (parson) to an ADUC_ConfigInfo object. We're not supposed to keep the root_value object around. It seems like there's a bug in the previous code where the root_value wasn't freed properly.
I'll free the root_value. Thanks for pointing that out.

Reply:
I updated the code to avoid unnecessary memory allocation, and stash some value in the parson json object where they make sense. Let me know if you have any more concerns.